### ISSUE TITLE
Major issues with trying to install onlyoffice integrated into nextcloud

### ISSUE THREAD

**Issue by phoenixz** [ONLYOFFICE/onlyoffice-nextcloud#463]
Hi there,

TL;DR; Trying to install onlyoffice integrated into next cloud, tried everything, nothing works. Please help, how can I get this installed without baning my head into a wall for 3 days straight?

I've tried multiple installations (by now way over 25 attempts over the span of 3 days)

All on Ubuntu server 21.04, all on one and the same server, all SSL with a real domain and letsencrypt. I want next cloud on default HTTPS port 443 and onlyoffice then in 444 so that to users it would be a seamless experience, they go to drive.mysmallcompany.com and get nextcloud, click on a document, and get into onlyoffice.

Next loud with letsencrypt was done withing 5 minutes.

I've tried installing onlyoffice-documentserver and onlyoffice-communityserver (individually and together) using apt. First off, those fail quietly if send mail already is installed, so make sure send mail is not installed. Those also fail quietly if apache already runs on port 80. Seriously, can you PLEASE add logging on failures? It changes a complete day of cursing into 30 seconds of "oh that is the problem? Lemme fix that real quick"

Then with apt document server: The document server will install and the health check shows "true", awesome! But next cloud integration keeps failing with the error that it's can't download the document. Yes, I've updated the "secret" and "header" AuthorizationJwt settings in onlyoffice local.json and nexcloud config.php, yes I've set not_peer_verify to true in next loud config.php. There are only errors logged in nextcloud and no errors or anything for that matter logged on onlyoffice, so I'm blind and have no idea what's wrong there. Again, LOG ERRORS! Also it seemed impossible to get this to run on any other port than 443, so i even moved nextcloud to port 444 which would have been a huge inconvenience for the users. Either way, couldn't get it to work.

Then with apt community server, health check shows a 502. Tracing that shows that the health check forwards to a service on port 9810, and the community edition apparently has no service on that port at all. Yes, i tried adding the ds-vpath, that just 404s, so the community edition appears to be completely useless or non functional

I've tried the docker installations which are headaches all on their own. At least my docker-fu improved a lot i guess? So i got that docker to work on external port 444 and to get the SSL working with letsencrypt. I go to the health check url and get a moving progress bar mentioning something about "The portal startup process might take some time, please wait" and "do not restart your computer".. yet, onlyoffice takes no CPU, no disk I/O, it's not doing anything it's not logging anything, no "initializing!" messages in any log file (I've checked them *all* in /app/onlyoffice/*), no access messages like you'd see in apache server access logs, no error messages. Nothing. I just get that progress bar with the "portal startup" message. Tracing that page showed me that it is a static page. I'm sure that some init process was the one showing it while it _should_ initialize, but why doesn't it do anything? Why doesn't it log anything? And isn't docker supposed to work out of the box? Why is this so hard..

That is about it. I've spent 3 days on a process that quite honestly should not have taken more than 5 minutes had there been only one install method. There are 5 different ones (one left to try) and all fail without any error indications which makes fixing the issue impossible.

There still is the option to use a snap installation, but quite honestly I don't want to anymore. If I go there, I'll again have to spend another day on fighting all the undocumented and unlogged errors that that snap installation will give me and I no longer have the energy.

Seriously.. Please please fix your installers pretty please. I know onlyoffice is big and complicated, but it cannot be this hard. I am an IT developer and system administrator with 25 years of experience and I cannot recall having encountered something so problematic to install as this. I was told beforehand that onlyoffice is awesome, but a horroshow to install. They were not wrong. If using onlyoffice is as great as the installation is horrible, then using it will be the best thing since sliced bread, which is the reason why I haven't given up yet. 

So in conclusion: What do I have to do, what goat do i have to sacrifice to get this to work?

**Comment by phoenixz** [comment]
![image](https://user-images.githubusercontent.com/1072698/117319189-b0b1ef80-ae50-11eb-9c7c-88b139d5d5f3.png)

**Comment by phoenixz** [comment]
Is there anybody out there that might know how I can install onlyoffice?

**Comment by tsunamaru** [comment]
@phoenixz personally, I suggest you install documentserver using docker, not apt. Process is fairly simple and eliminates your deal with dependencies and various OS configs.
https://helpcenter.onlyoffice.com/en/installation/docs-community-install-docker.aspx
